### DISCUSSION TITLE
windows wheel requires `python3.dll` but it's not available for non default python interpreter.

### DISCUSSION THREAD

**Discussion by ghyatzo** [PyO3/maturin#2223] [Q&A] [answered]
Hello,

I'm building a mixed python rust project which I manage to build just fine on windows (natively). The resulting wheel works correctly when installed and called inside the normal python interpreter.
But I would also require this wheel to be loaded from the python interpreter of `sublime text`, which is the editor plugin host.
But on windows, when installing the wheel for sublime and then trying to import it, it fails with a
`ImportError: DLL load failed while importing codemp: ...` and it seems it's trying to call `<ST>/python3.dll` which doesn't exists since Sublime text does not provide the `.dll` (`<ST>` is the Sublime Text installation directory).
The same project built on both linux and macos works correctly in the native interpreter and the Sublime Text one.

Is there a way to circumvent this? I've never dealt with python on windows this deep so I am not really sure what is going on, or if it even makes sense to try and statically link the library directly? I've noticed that for the cross-compilation to windows there are some shenanigans involving this library but I am unsure if that could be useful or not.
I would be eternally grateful for any help

**Comment by messense** [comment, marked as answer]
Sublime Text only provides `python33.dll` and `python38.dll` thus it doesn't support `abi3`, so you can't use `abi3` wheels with it, you must build a python version specific wheel.

**Reply by ghyatzo** [reply]
That's great news thanks for the quick reply.
Why is the abi3 not an issue for sublime on Linux/macos ?

**Reply by messense** [reply]
Because Windows sucks?

See https://peps.python.org/pep-0513/#libpythonx-y-so-1 and https://peps.python.org/pep-0384/#linkage

**Reply by messense** [reply]
Sublime Text should really include a `python3.dll` in its distribution, but it can be complicate because it can use both python3.3 and python3.8, so to provide a stable abi for both of them, the `python3.dll` may only contain symbols from python3.3 which may not work with `pyo3` (though I'm not sure because I don't develop on Windows).

**Reply by ghyatzo** [reply]
Amen, thanks for the insights!
I guess I'll just treat windows as the special child once again and offer an extra wheel just for him.
